index.js

```diff
@@ -8,7 +8,27 @@ const token = envy().token;
 let playerList = [];
 let captainList = [];
 
-const prefix = '!'
+const prefix = '!';
+
+//abstracting addPlayer will let us use it in other scenarios, esp. automated testing.
+function addPlayer(player, msg = undefined) {
+    if (playerList.includes(player)) {
+        if (msg != undefined) { msg.reply(`You are already queued up! If you would like to leave, please use ${prefix}leave`); };
+        return [...playerList];
+    };
+    return [...playerList, player];
+};
+
+//abstracting join lets us edit and see it easier, instead of inside the tangle of ifs in the callback. 
+function join(msg) {
+    msg.delete();
```

Comment:
Assuming we'll have a separate channel for queue sniping and bot spam, imo it's a good idea to keep the original command message there for the sake of not confusing the user as the channel will be spammed with commands anyways.